db_test.go

```diff
@@ -37,6 +30,45 @@ func TestOpen(t *testing.T) {
 	ok(t, db.Close())
 }
 
+// Ensure that opening a database with a bad path returns an error.
+func TestOpen_BadPath(t *testing.T) {
+	for _, path := range []string{
+		"",
+		filepath.Join(tempfile(), "youre-not-my-real-parent"),
+	} {
+		t.Logf("path = %q", path)
+		db, err := bolt.Open(path, 0666, nil)
+		assert(t, err != nil, "err: %s", err)
+		equals(t, path, err.(*os.PathError).Path)
+		equals(t, "open", err.(*os.PathError).Op)
+		equals(t, (*bolt.DB)(nil), db)
+	}
+}
+
+// Ensure that opening a file with wrong checksum returns ErrChecksum.
+func TestOpen_ErrChecksum(t *testing.T) {
+	t.Skip("pending")
```

Comment:
You can create a meta page like this:

```
buf := make([]byte, 4096)
var meta := (*struct {
	magic    uint32
	version  uint32
	_        uint32
	_        uint32
	_        [16]byte
	_        uint64
	_        uint64
	_        uint64
	checksum uint64
})(unsafe.Pointer(&buf[0]))
```

And then set the fields:

```
// Here's an invalid version
meta.magic = 0xED0CDAED
meta.version = 100
```

```
// Here's an invalid checksum
meta.magic = 0xED0CDAED
meta.version = 2
meta.checksum = 123
```

Then write the `buf` to a file and try to open it.